### ISSUE TITLE
Gradient with respect to input is `None`

### ISSUE THREAD

**Issue by akanksha-atrey** [openai/baselines#1001]
I am trying to calculate the gradient of the outputs with respect to the input but I am getting None consistently. I am attempting this in `baselines/baselines/common/policies.py` in the following function (note `tf.gradients(a_logits, [observation]))`).

```
def step(self, observation, **extra_feed):
    a, a_logits, v, state, neglogp = self._evaluate([self.action, self.pi, self.vf, self.state, self.neglogp], observation, **extra_feed)
    if state.size == 0:
        state = None

    print(tf.gradients(a_logits, [observation]))
    return a, v, state, neglogp, a_logits.flatten()
```

Has anyone faced this before? Note, I am getting [None] output even for gradient w.r.t to the negative log probability (neglogp). I would really appreciate any help!

**Comment by christopherhesse** [comment]
Is the observation a differentiable set of parameters here?
